src/server/tiering/external_alloc.cc

```diff
@@ -479,11 +496,19 @@ void ExternalAllocator::FreePage(Page* page, SegmentDescr* owner, size_t block_s
 
   // Remove fast allocation reference.
   if (free_pages_[bidx] == page) {
-    free_pages_[bidx] = &empty_page;
+    free_pages_[bidx] = page->next_free ? page->next_free : &empty_page;
```

Comment:
I am not sure if `page->next_free` can ever be null in this case. 
I thought free_pages_ list always terminates with `&empty_page`.

Reply:
no, empty_page ends with nullptr. I don't see a reason to fight with it because it allows us easily to place empty_page into the free_pages_ slot without additional checks